src/main/java/receiver/BmsReceiver.java

```diff
@@ -0,0 +1,75 @@
+package main.java.receiver;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+public class BmsReceiver {
+
+	/**
+	 * @param args
+	 */
+	public static void main(String[] args) {
+		System.out.println("----- Reading input from sender started -------");
+		HashMap<String, List<Double>> paramMap = new HashMap<>();
+		try (BufferedReader reader = new BufferedReader(new InputStreamReader(System.in))) {
+			// to create the list of parameter names coming from sender
+			List<String> parametersList = Arrays.asList(reader.readLine().split(";"));
+			paramMap = prepareParamMap(parametersList);
+			readAndPerformOpronDataFromSender(reader, parametersList, paramMap);
+		} catch (Exception e) {

```

Comment:
This will catch every exception, including the ones you dont expect in the normal flow. Sometimes this makes it hard to debug, like Null-pointer exceptions, etc. Catch specific exceptions, only those you expect to be thrown by the code.